### PR TITLE
[BugFix] Orc reader will set data even if it exceeds varchar's length

### LINKED ISSUE #10370: broker load from orc cannot pass check_string_lengths
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE `mtable` (
  `a` bigint(20) NULL COMMENT " ",
  `b` varchar(12) NULL COMMENT " ",
  `c` varchar(12) NULL COMMENT ""
) ENGINE=OLAP 
DUPLICATE KEY(`a`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`a`) BUCKETS 1 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```
2. prepare data, transfer it into orc
```
1,123456789012,123456
2,12345678901,12345
```
3. BROKER LOAD '....'
```
LOAD LABEL orc_load2 (DATA INFILE("xxx/a.orc") INTO TABLE `mtable`) 
WITH broker hdfs_broker PROPERTIES("timeout"="300","max_filter_ratio"="0.0");
```

### Expected behavior (Required)
Load success, like csv
### Real behavior (Required)
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.1.12

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10370

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

Unlike csv scanner, orc_scanner_adapter will set value even data's len exceeds varchar's len.
This will lead to check_string_lengths fails.

### CODE REVIEW

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 1]
```diff
@@ -571,14 +571,22 @@ static void fill_string_column(orc::ColumnVectorBatch* cvb, ColumnPtr& col, int
     if (type_desc.type == TYPE_CHAR) {
         // Possibly there are some zero padding characters in value, we have to strip them off.
         for (int i = col_start; i < col_start + size; ++i, ++pos) {
-            size_t str_size = remove_trailing_spaces(data->data[pos], data->length[pos]);
-            vb.insert(vb.end(), data->data[pos], data->data[pos] + str_size);
-            vo.emplace_back(vb.size());
+            if (LIKELY(data->length[pos] <= type_desc.len)) {
```
Comment (chaoyli) [inline be/src/formats/orc/orc_chunk_reader.cpp:574]: If the data->length[pos] > type_desc.len, it still the same?

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request closed without merging**